roomserver/alias/alias.go

```diff
@@ -46,6 +47,12 @@ type RoomserverAliasAPIDatabase interface {
 	// Remove a given room alias.
 	// Returns an error if there was a problem talking to the database.
 	RemoveRoomAlias(ctx context.Context, alias string) error
+	// Return the room NID for a room ID.
+	RoomNID(ctx context.Context, roomID string) (types.RoomNID, error)
+	// Look up the room version for a given room.
+	GetRoomVersionForRoom(
+		ctx context.Context, roomNID types.RoomNID,
```

Comment:
Nowhere else in the interface do we pass in room NIDs, why do that here? Leave it as an implementation detail.

Reply:
I've stripped NIDs from the interface.